### PR TITLE
Use sha instead of ref for PRs when possible.

### LINKED ISSUE #1658: Can't Fetch Renovate File For Validation on PR from Fork
#### This is a:

* [x] Bug report (non-security related)
* [ ] Feature request
* [ ] I'm not sure which of those it is

#### I'm using:

* [ ] The Renovate GitHub App
* [x] Self-hosted GitHub
* [ ] Self-hosted GitLab
* [ ] Self-hosted VSTS

#### Please describe the issue:

I recently upgraded an on-premise Renovate instance to version [11.36.2](https://github.com/renovateapp/renovate/releases/tag/11.36.2) and I'm now receiving the following log message: `WARN: Error checking PR (repository=org/project)`

I recently filed a pull request against the `org/project` repository with a change that modifies the contents of the `renovate.json` file.

My pull request is from a fork of the `org/project` repository called `hbetts/project`, and the branch is called `patch-1`.

Renovate successfully determines that my pull request contains a modified `renovate.json` configuration file using the [following code](https://github.com/renovateapp/renovate/blob/862998c2b43c8d5b9dfd54e3d81ebeb221e19bae/lib/workers/repository/validate.js#L30).

When it [attempts to fetch the `patch-1` branch](https://github.com/renovateapp/renovate/blob/862998c2b43c8d5b9dfd54e3d81ebeb221e19bae/lib/workers/repository/validate.js#L39) it uses the following path - `/api/v3/repos/org/project/git/refs/heads/patch-1`

That path is for the upstream repository, not my fork. Therefore the request fails, as the branch, `patch-1`, is not on the upstream repository.

### PR DESCRIPTION
**Pull request by kayoub5**
Resolves #1658
@destroyerofbuilds could you check if this fixes the problem for you?

### CODE REVIEW

**Comment by destroyerofbuilds** [comment]
@kayoub5 how does using a `sha` reference resolve #1658?

File: test/platform/github/__snapshots__/index.spec.js.snap [hunk 1]
```diff
@@ -456,6 +456,7 @@ Object {
   "isUnmergeable": true,
   "mergeable_state": "dirty",
   "number": 1,
+  "sha": undefined,
```
Comment (destroyerofbuilds) [inline test/platform/github/__snapshots__/index.spec.js.snap]: Is there a way to test a scenario in which `sha` isn't `undefined`?

**Comment by rarkins** [comment]
@destroyerofbuilds GitHub does some wicked things with git behind the scenes. For instance, notice how the full details of a PR are still available forever, even after the fork that submitted them has been deleted. What we've found is that forks are almost like "the same repository" and definitely not a full clone, and SHAs are visible to the source repo once submitted as a PR.

**Comment by destroyerofbuilds** [comment]
> SHAs are visible to the source repo once submitted as a PR.

Okay, so that definitely helps me understand why this particular change addresses the visibility issue reported in #1658.

I'd love to check out this fix on our deployment. I'll do that as soon as I get a chance to deploy a new release related to #1775

**Comment by destroyerofbuilds** [comment]
👍 Looks good to me!

I cherry-picked this commit into a local test environment, and I can confirm it solved the error we were encountering in #1658 

Before the change:

```bash
DEBUG: GET repos/org/repo/contents/renovate.json?ref=hbetts-patch-1 [retries=5] (repository=org/repo)
INFO: getFile 404 (repository=org/repo
       "filePath": "renovate.json",
       "branchName": "hbetts-patch-1"
```

After the change:

```bash
DEBUG: GET repos/org/repo/contents/renovate.json?ref=31857723b76a543e3fbb136ae280f56e37af8f2c [retries=5] (repository=org/repo)
DEBUG: Validating config (repository=org/repo)
       "config": {
         "automerge": true,
         "major": {"automerge": false},
         "labels": ["refactor"],
         "pinVersions": false,
         "rebaseStalePrs": true,
         "unpublishSafe": true
       }
```

**Pull request merged**